include/blas_meta.h

```diff
@@ -159,6 +160,13 @@ first_vector_t concatenate_vectors(first_vector_t first_vector,
   return std::move(first_vector);
 }
 
+/**
+ * @brief Defines if a type is a scalar within the context of sycl.
+ * @tparam type The type to be tested.
+ */
+template <typename type>
```

Comment:
This is still using a C++17 feature (template variables). It compiles but give a warning. 
I seem to be unable to get it working with C++14 (other than copying the condition to all the function declarations)
Open to suggestions :)

Reply:
What is the warning exactly? `std::is_scalar` and `std::conditional` both seem to be present in C++11.

Reply:
The code already has the changes suggested by @hjabird . Before I was using variable templates which are C++17 only feature